### ISSUE TITLE
Infinite loop in `parse::find_port`

### ISSUE THREAD

**Issue by xiaozhiyan** [timothee-haudebourg/iref#20]
Hi, thanks for releasing `iref` v3.0.0.
I'm trying to use the new version in [hadoop-rust](https://github.com/innorealm/hadoop-rust), but I've found that the function [`Authority::port`](https://github.com/timothee-haudebourg/iref/blob/3.0.0/src/iri/authority.rs#L60) does not return.
The issue seems to be caused by an infinite loop in [`parse::find_port`](https://github.com/timothee-haudebourg/iref/blob/3.0.0/src/common/parse.rs#L274).
Please help to fix it when convenient.

(Here's a temporary fix for it.)
```
% git diff src/common/parse.rs
diff --git a/src/common/parse.rs b/src/common/parse.rs
index 3980802..8eb90ab 100644
--- a/src/common/parse.rs
+++ b/src/common/parse.rs
@@ -276,9 +276,10 @@ pub fn find_port(bytes: &[u8], mut i: usize) -> Option<Range<usize>> {
                                        i += 1;
                                        continue 'host;
                                }
+                               i += 1;
                        }

-                       return Some(start..i);
+                       return Some(start..i - 1);
                }

                i += 1
```

**Comment by timothee-haudebourg** [comment]
Thanks for the report, I'm missing some tests there. I've fixed the function, I'll just cover every function of the module with tests and update the crate.

**Comment by timothee-haudebourg** [comment]
Fix published with version 3.0.1
